File: server/src/main/java/com/raczy/chinesecheckers/Field.java
package com.raczy.chinesecheckers;

import java.util.Arrays;

/**
 * Graph Node class holding info about player's checker presence and references to neighbours.
 * Created by kacperraczy on 12.12.2017.
 */
public class Field {
    public final static int TOPRIGHT = 0;
    public final static int RIGHT = 1;
    public final static int BOTTOMRIGHT = 2;
    public final static int BOTTOMLEFT = 3;
    public final static int LEFT = 4;
    public final static int TOPLEFT = 5;

    public final static int NEIGHBOUR_COUNT = 6;

    private final int id;
    private Field[] neighbours;
    private Player player = null;

    public Field(final int id) {
        this.id = id;

        Field[] neighbours = new Field[NEIGHBOUR_COUNT];
        Arrays.fill(neighbours, null);
        this.neighbours = neighbours;
    }

    public int getId() {
        return id;
    }

    public Field[] getNeighbours() {
        return neighbours;
    }

    public Player getChecker() {
        return player;
    }

    public void setChecker(Player player) {
        this.player = player;
    }

    public boolean isOccupied() {
        return this.getChecker() != null;
    }

    public static int oppositeEdge(int eno) {
        return (eno + 3) % NEIGHBOUR_COUNT;
    }

}

File: client/ChineseCheckers/ChineseCheckers Shared/Logic/Field.swift
//
//  Field.swift
//  ChineseCheckers
//
//  Created by Kacper Raczy on 26.12.2017.
//  Copyright © 2017 Kacper Raczy. All rights reserved.
//

import Foundation

class Field {
    enum Direction: Int {
        case topRight = 0
        case right
        case bottomRight
        case bottomLeft
        case left
        case topLeft
        
        
        func opposite() ->Direction {
            let val = (self.rawValue + 3) % 6
            let dir = Direction(rawValue: val)!
    
            return dir
        }
        
        static let allValues: [Direction] = [topRight, right, bottomRight, bottomLeft, left ,topLeft]
    }
    
    private let _id: Int
    private var _neighbours: [Field?]
    private var _player: Player?
    
    init(id: Int) {
        self._id = id
        self._neighbours = [Field?](repeating: nil, count: 6)
    }
    
    convenience init(_ id: Int, player: Player?) {
        self.init(id: id)
        self.player = player
    }
    
    var id: Int {
        return self._id
    }
    
    var neighbours: [Field?] {
        return self._neighbours
    }
    
    var player: Player? {
        get{
            return _player
        }
        set{
            self._player = newValue
        }
    }
    
    var isOccupied: Bool {
        return self._player != nil
    }
    
    func getNeighbour(dir: Direction) -> Field? {
        return self._neighbours[dir.rawValue]
    }
    
    func setNeighbour(dir: Direction, field: Field?) {
        self._neighbours[dir.rawValue] = field
    }
    
    
}
